readme update (#6)
eclipse-thingweb/ui-wot: README.md

# UI-WoT

> ## 👋 Hey there!

Reviewer: no need for title

> Welcome to **UI-WoT**, a Google Summer of Code 2025 project with Eclipse Foundation!. We are working to parse any valid WoT Thing Description, extract all its affordances (properties, actions, and events), and then automatically render them as usable UI widgets, and enable real-time interaction

Reviewer: We don't need to mention that it is a GSoC project

Reviewer: Also it is a UI generator and a library to embed these widgets in different web front ends.

## 🎯 What We Aim

By the end of this GSoC journey, UI-WoT will be able to:

*   **Consume Any TD:** Load and understand any valid W3C WoT Thing Description.
*   **Auto-Generate UI:** Instantly create interactive UI cards for every property, action, and event in that TD.
*   **UI Choices:** Give you options to pick from many UI for each affordance (e.g., a slider for a number, a toggle for a boolean).
*   **Personalized Dashboards:** Let you drag, drop, and resize cards to build your perfect control panel, and even save your layouts.

## 📑 Flow Chart 
![UI-WoT Flow Chart](./public/images/flowchart.png)

## </> The Proposed Tech

These are currently proposed tools, may changes according to further need. 

| Tool | Purpose | Reason |
|---|---|---|

> | **Vite** | Build Tool, Development Server | Fast development server, quick build times, optimized for modern web development.|

Reviewer: We did not agree on this but I guess it is fine?

| **React** | Frontend Framework | Component-based architecture, large community, ideal for dynamic UI generation.|
| **Tailwind CSS** | CSS Framework | Utility-first approach good for rapid UI development, highly customizable, excellent for responsive design.|

> | **DaisyUI & Others** | Component Library | Pre-styled accessible UI components on top of Tailwind CSS will enhance UI development.|

Reviewer: Did we agree on this?

> | **Node-WoT** | WoT Interaction Library | Official W3C WoT implementation for JavaScript, provides high-level API for interacting with Things.|

Reviewer: ```suggestion
| **Node-WoT** | WoT Interaction Library | Reference W3C WoT implementation for JavaScript, provides high-level API for interacting with Things.|
```

| **Vitest** | Testing Framework | Since we're on Vite it is natural fit, Jest-compatible. |
| **React Grid Layout** | Dashboard Layout | For drag and drop and resizable grid layout for building interactive dashboards. |

## 💠 Figma Design
Feel free to give suggestions here, will love to hear from you!
[View the UI-WoT Figma Design](https://www.figma.com/design/PcgN3oVPl6387MqqidCF1H/UI-WOT?node-id=0-1&t=6rQYTFoWH27GqBwQ-1)

> I will be sharing the progress with the community. Stay tuned!

Reviewer: ```suggestion
We will be sharing the progress with the community. Stay tuned!
```

</picture>
</h1>

> > A library, reusable component, and dashboard tool to interact with any WoT-compliant Thing Description (TD).

Reviewer: ```suggestion
> A library, reusable component, and dashboard tool to interact with any IoT device with a WoT Thing Description (TD).
```

The Eclipse Thingweb **UI-WoT** is a modular toolkit being designed to serve mutliple use cases. It follows the W3C Web of Things standards, ensuring high interoperability. With UI-WoT, developers can skip manual UI coding and instead generate dynamic, interactive dashboards using standardized TDs.

## What We Aim

UI-WoT is being designed to be used in **three different ways**, depending on your needs:

### 1. Reusable UI Components

Web Components (e.g., `<ui-toggle>`, `<ui-slider>`, `<ui-button>`) that can be used in any web project to represent common WoT affordances like Properties, Actions, and Events. These components:

* Will be implemented as **standard Web Components** (Custom Elements).
* Will work in any framework (React, Vue, Angular) or plain HTML.
* Encapsulated interaction logics like state sync, event listening, property writing etc.

> This makes these ideal to use in any custom dashboards, prototyping tools, or in embedded UIs where you want to control individual device affordances without managing low-level WoT TD interaction.

Reviewer: ```suggestion
This makes our components easy to use in any custom dashboards, prototyping tools, or in embedded UIs where you want to control individual device affordances without managing low-level WoT TD interaction.
```

### 2. Embeddable UI Generation Library

UI-WoT can be imported as a JavaScript library into existing applications to dynamically generate user interfaces from any valid TD. By calling this library developers can:

* Parse the TD structure and extract all affordances.
* Attach the UI to any custom container or layout.

This approach will be useful for those who want to integrate dynamic WoT device control into their existing apps or dashboards with minimal effort and maximum flexibility.

### 3. UI Dashboard Generator

> UI-WoT will also include a fully functional web application that a huge upgrade to 'browserified node-wot':

Reviewer: ```suggestion
UI-WoT will also include a fully functional web application that is a significant upgrade to the example UI of node-wot's browser bundle, i.e. the 'browserified node-wot':
```
